src/routing/history/StateHistory.ts

```diff
@@ -14,26 +28,26 @@ export class StateHistory implements HistoryInterface {
 		this._onChangeFunction = onChange;
 		this._window = window;
 		this._base = base;
+		if (!trailingSlash.test(this._base)) {
+			this._base = `${this._base}/`;
+		}
 		this._current = this._window.location.pathname + this._window.location.search;
 		this._window.addEventListener('popstate', this._onChange, false);
 		this._onChange();
 	}
 
 	public prefix(path: string) {
-		const baseEndsWithSlash = /\/$/.test(this._base);
-		const pathStartsWithSlash = /^\//.test(path);
-		if (baseEndsWithSlash && pathStartsWithSlash) {
-			return this._base + path.slice(1);
-		} else if (!baseEndsWithSlash && !pathStartsWithSlash) {
-			return `${this._base}/${path}`;
-		} else {
-			return this._base + path;
+		if (path[0] === '#') {
+			path = path.slice(1);
+		}
+		if (path[0] === '/') {
```

Comment:
this can be one if with the above right?

Reply:
I did it this was to guard against `#/link` which needs to be `link`